include/ofi_mr.h

```diff
@@ -226,6 +226,9 @@ struct ofi_mr_entry {
 	unsigned int			subscribed:1;
 	int				use_cnt;
 	struct dlist_entry		list_entry;
+#ifdef HAVE_LIBCUDA
+	uint64_t			cuda_buf_id;
+#endif
```

Comment:
We'll want a separate flag or specific id value to indicate if this field is invalid, so we can skip the cuda check on cache lookups for non-cuda buffers.